src/exception/validation.ts

```diff
@@ -41,10 +41,12 @@ export const ValidationResultType = {
 
 type ValidationResultErrorType = ObjectValues<typeof ValidationResultType>;
 
-type ValidationError = {
+export type ValidationError = {
+  message: string;
+  result: ValidationResultErrorType;
   field: string;
   recordIndex: number;
-} & ValidationResult;
+};
```

Comment:
not a request for change: what's the benefit of spelling these properties out? it seems as if the `&` did the same job.

Reply:
changed usage of types so they're separate distinct entities. 
they share those fields but ValidationError isn't linked to ValidationResult, in the sense that we don't want a change to ValidationResult changing the ValidationError type.

a small code smell in the structure of the validation overall is what is really surfacing here.

ValidationResult's come from individual validator functions
ValidationError's come from the overall validate that runs the individual validators

Reply:
in that case, both of those different types you mention could extend/inherit from a shared base type, no?